Include/methodobject.h

```diff
@@ -13,7 +13,11 @@ extern "C" {
 
 PyAPI_DATA(PyTypeObject) PyCFunction_Type;
 
-#define PyCFunction_Check(op) (Py_IS_TYPE(op, &PyCFunction_Type) || (PyType_IsSubtype(Py_TYPE(op), &PyCFunction_Type)))
+#define PyCFunction_CheckExact(op) Py_IS_TYPE(op, &PyCFunction_Type)
```

Comment:
I would prefer to exclude PyCFunction_CheckExact() from the limited C API.

Reply:
I'm not so sure. We are changing the semantics of `PyCFunction_Check()` by adding a subtype that it covers. The macro is a part of the limited C-API and thus, the semantic change is a problem all by itself. Either we accept that, then we should add the "obvious" related `CheckExact` macro, also to the limited API. Or, we reject that change all together. I don't think the latter is an option, though.

Personally, I'd say, since it's a macro, it doesn't hurt, since it has no impact at all on the ABI. Everyone can safely copy and use that macro also on their side, so why not just add it?

Reply:
I will not hold this PR just for that. We can should PyCFunction_CheckExact() as it is in Python 3.9, and revisit the issue later. I created https://bugs.python.org/issue40601 to explain my concern: I plan to work on that in Python 3.10, it's too late for Python 3.9.

Reply:
Do note that `PyCMethod` is a bit of a hack in that `PyCFunction` is still not an "acceptable base type". You can't subclass`type(print)`, but static types happen to bypass the `Py_TPFLAGS_BASETYPE` check. This is one more thing to be resolved if we want all types can become heap types.